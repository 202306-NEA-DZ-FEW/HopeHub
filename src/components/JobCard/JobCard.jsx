// import { useTranslation } from "next-i18next";
import React from "react";
import { TbDeviceImacCode } from "react-icons/tb";
import { CiLocationOn } from "react-icons/ci";
function JobCard({ title, type, location, tags, category }) {
    // const { t } = useTranslation("common");
    return (
        <div className='shrink-0 shadow-[5px_12px_12px_0px_rgba(0,0,0,0.61)] rounded-[11px] p-4'>
            <div className='flex gap-2 py-3'>
                <TbDeviceImacCode></TbDeviceImacCode>
                <span>{category}</span>
            </div>
            <div className=' font-bold py-3'>
                <span>
                    {title} | {tags} |{type}{" "}
                </span>
            </div>
            <div className='flex gap-2 py-3'>
                <CiLocationOn></CiLocationOn>
                {location}
            </div>
        </div>
    );
}

export default JobCard;
