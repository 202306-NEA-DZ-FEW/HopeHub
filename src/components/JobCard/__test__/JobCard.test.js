import renderer from "react-test-renderer";

import JobCard from "../JobCard";

it("renders correctly", () => {
    const tree = renderer.create(<JobCard />).toJSON();
    expect(tree).toMatchSnapshot();
});
