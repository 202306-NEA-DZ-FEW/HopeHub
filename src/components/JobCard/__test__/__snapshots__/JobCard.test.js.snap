// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="bg-NeutralWhite w-36 h-44 md:w-56 md:h-28 lg:w-96 lg:h-32 dark:bg-Dark_Primary rounded-md shadow-md p-4 hover:shadow-lg transform transition-transform duration-300 hover:scale-105 cursor-pointer overflow-hidden"
>
  <div
    className="flex items-center gap-2 my-2 "
  >
    <svg
      className="text-gray-500 dark:text-gray-400"
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
        stroke="none"
      />
      <path
        d="M11.5 17h-7.5a1 1 0 0 1 -1 -1v-12a1 1 0 0 1 1 -1h16a1 1 0 0 1 1 1v9"
      />
      <path
        d="M3 13h18"
      />
      <path
        d="M8 21h3.5"
      />
      <path
        d="M10 17l-.5 4"
      />
      <path
        d="M20 21l2 -2l-2 -2"
      />
      <path
        d="M17 17l-2 2l2 2"
      />
    </svg>
  </div>
  <div
    className="font-bold h-2/4 md:h-1/4 my-2 overflow-hidden"
  >
    <span
      className="text-lg text-black dark:text-white"
    />
  </div>
  <div
    className="flex items-center gap-2 py-2"
  >
    <svg
      className="text-gray-500 dark:text-gray-400"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        id="Location_On"
      >
        <g>
          <path
            d="M12,21.933a1.715,1.715,0,0,1-1.384-.691L5.555,14.5a7.894,7.894,0,1,1,12.885-.009L13.385,21.24A1.717,1.717,0,0,1,12,21.933ZM11.992,3.066A6.81,6.81,0,0,0,7.414,4.815a6.891,6.891,0,0,0-1.05,9.1l5.051,6.727a.725.725,0,0,0,.584.292h0a.732.732,0,0,0,.586-.292l5.044-6.734A6.874,6.874,0,0,0,12.81,3.113,7.277,7.277,0,0,0,11.992,3.066Z"
          />
          <path
            d="M12,12.5A2.5,2.5,0,1,1,14.5,10,2.5,2.5,0,0,1,12,12.5Zm0-4A1.5,1.5,0,1,0,13.5,10,1.5,1.5,0,0,0,12,8.5Z"
          />
        </g>
      </g>
    </svg>
    <span
      className="text-sm text-gray-600 dark:text-gray-300  truncate"
    />
  </div>
</div>
`;
